source/network/ServerSideNetworkHandler.cpp

```diff
@@ -520,3 +522,49 @@ void ServerSideNetworkHandler::commandTP(OnlinePlayer* player, const std::vector
     
 	sendMessage(player, ss.str());
 }
+void ServerSideNetworkHandler::commandGive(OnlinePlayer * player, const std::vector<std::string>&parms)
+{
+	if (!m_pLevel)
+		return;
+	if (player->m_pPlayer != this->m_pMinecraft->m_pLocalPlayer)
+	{
+		sendMessage(player, "Sorry, only the host can use this command at the moment");
+		return;
+	}
+	if (parms.size() != 2)
+	{
+		sendMessage(player, "Usage: /give [item] [item count]");
+		return;
+	}
+	int id = 0;
+	int amount = 1;
+	if (!sscanf(parms[0].c_str(), "%d", &id) || !sscanf(parms[1].c_str(), "%d", &amount))
+	{
+		sendMessage(player, "Usage: /give [item] [item count]");
+		return;
+	}
+
+	Inventory* pInventory = player->m_pPlayer->m_pInventory;
+
+	pInventory->addTestItem(id, amount);
```

Comment:
Couldn't you check if `Item::items[id]` is null before this to prevent crashes? Although I'm not sure if Item::items is zeroed beforehand in MCPE (in MCPI it mostly is, except for a single special case, and in MCJE it is).

Reply:
I'll try that when I'm back home in about 4 hours, thankyou

Reply:
Added a check for that as well as if the id is greater than 512 or less than 0, resolving the possibility of a crash